torch/hub.py

```diff
@@ -473,6 +490,24 @@ def load_state_dict_from_url(url, model_dir=None, map_location=None, progress=Tr
     cached_file = os.path.join(model_dir, filename)
     if not os.path.exists(cached_file):
         sys.stderr.write('Downloading: "{}" to {}\n'.format(url, cached_file))
-        hash_prefix = HASH_REGEX.search(filename).group(1)
-        _download_url_to_file(url, cached_file, hash_prefix, progress=progress)
+        hash_prefix = HASH_REGEX.search(filename).group(1) if check_hash else None
+        download_url_to_file(url, cached_file, hash_prefix, progress=progress)
+
+    # Note: extractall() defaults to overwrite file if exists. No need to clean up beforehand.
+    if tarfile.is_tarfile(cached_file):
```

Comment:
We can maybe factor this out in a function in a follow up PR, so that it can be used by other libs as well, see for example https://github.com/pytorch/vision/blob/a129b6b86a75f2d1fc80055f2cd0fae63efc0d2d/torchvision/datasets/utils.py#L216

Reply:
Yea agree, I was also thinking about maybe we can expose this as an API and let `torch.load` handle compressed file directly :D